src/ooxml/java/org/apache/poi/xddf/usermodel/chart/XDDFLineChartData.java

```diff
@@ -67,7 +67,11 @@ public Grouping getGrouping() {
     }
 
     public void setGrouping(Grouping grouping) {
+        if (chart.getGrouping()!=null) {
         chart.getGrouping().setVal(grouping.underlying);
+        } else {
+            chart.addNewGrouping().setVal(grouping.underlying);
+        }
```

Comment:
If you were caught by this when using it in your own code, then you should also consider to apply the (correctly formatted) fix to other chart data classes which have the `setGrouping` defined on them.

Reply:
updated code